layers/+lang/c-c++/funcs.el

```diff
@@ -182,3 +185,161 @@ and the arguments for flyckeck-clang based on a project-specific text file."
+    (progn (condition-case nil
+             (spacemacs//c-c++-lsp-call-function "lsp-" "-enable")
+             (user-error nil))))
+
+(defun spacemacs//c-c++-lsp-config ()
+  "Configure the LSP backend specified by the `c-c++-backend' configuration variable."
+    (progn
+      (spacemacs//c-c++-lsp-setup-company)
+      (spacemacs//c-c++-lsp-customise-lsp-ui-peek)
+      (spacemacs//c-c++-lsp-wrap-functions)
+      (setq-default flycheck-disabled-checkers '(c/c++-clang c/c++-gcc))
+
+      (if (eq c-c++-lsp-cache-dir nil)
+        (progn
+          (setq c-c++-lsp-cache-dir (file-truename(concat "~/.emacs.d/.cache/" (symbol-name c-c++-backend))))
+          (message (concat "c-c++: No c-c++-lsp-cache-dir specified: defaulting to " c-c++-lsp-cache-dir))))
+
+      (dolist (param '("executable" "extra-init-params" "cache-dir" "project-whitelist" "project-blacklist" "sem-highlight-method"))
+        (spacemacs//c-c++-lsp-apply-config param))
+
+      (when c-c++-lsp-sem-highlight-rainbow
+        (unless c-c++-lsp-sem-highlight-method
+          (progn
+            (setq c-c++-lsp-sem-highlight-method 'font-lock)
+            (message "c-c++: No semantic highlight method specified. Defaulting to `font-lock'.")))
+        (ecase c-c++-backend
+          ('lsp-cquery (cquery-use-default-rainbow-sem-highlight))
+          ('lsp-ccls (ccls-use-default-rainbow-sem-highlight))))
+
+      (dolist (mode c-c++-modes)
+        (spacemacs/lsp-bind-keys-for-mode mode)
+        (spacemacs//c-c++-lsp-bind-keys-for-mode mode))
+
+      (evil-set-initial-state '(spacemacs//c-c++-lsp-symbol nil "-tree-mode") 'emacs)
+      ;;evil-record-macro keybinding clobbers q in cquery-tree-mode-map for some reason?
+      (evil-make-overriding-map (symbol-value (spacemacs//c-c++-lsp-symbol nil "-tree-mode-map")))))
+
+(defun spacemacs//c-c++-lsp-setup-company ()
+  "Setup LSP backend auto-completion."
+    (progn
+      (spacemacs|add-company-backends :backends company-lsp :modes c-mode-common)
+      ;;Disable client-side cache and sorting, as server does a better job
+      (setq company-transformers nil company-lsp-async t company-lsp-cache-candidates nil)))
+
+(defun spacemacs//c-c++-lsp-wrap-functions ()
+  "Wrap navigation functions for the LSP backend specified by the `c-c++-backend' configuration variable."
+  (defun c-c++/call-hierarchy () (interactive) (spacemacs//c-c++-lsp-funcall-interactively nil "-call-hierarchy" nil))
+  (defun c-c++/call-hierarchy-inv () (interactive) (spacemacs//c-c++-lsp-funcall-interactively nil "-call-hierarchy" t))
+  (defun c-c++/inheritance-hierarchy () (interactive) (spacemacs//c-c++-lsp-funcall-interactively nil "-inheritance-hierarchy"))
+  (defun c-c++/inheritance-hierarchy-inv () (interactive) (spacemacs//c-c++-lsp-funcall-interactively nil "-inheritance-hierarchy" t))
+  (defun c-c++/member-hierarchy () (interactive) (spacemacs//c-c++-lsp-funcall-interactively-no-args nil "-member-hierarchy"))
```

Comment:
memberHierarchy has been deleted from cquery.

Reply:
:) Sounds like it may be reinstated:
https://github.com/cquery-project/cquery/issues/789
I'll hang tough for a few days and see what happens

Reply:
... and it seems to have been reinstated